contents/security/sqlInjection.md

```diff
@@ -13,45 +13,87 @@
 
 # SQL Injection
 
-Author: [Lewis Koh](https://github.com/nus-cs3281/2018/blob/master/students/lewisKoh/lewisKoh-Resume.md)
+Author: [Lewis Koh](https://github.com/nus-cs3281/2018/blob/master/students/lewisKoh/lewisKoh-Resume.md), [Jiang Chunhui](https://github.com/Adoby7)
 
-## Overview
+## What is SQL Injection?
 
-SQL (Structured Query Language) Injection is a vulnerability found in websites using databases.
-
-SQL is a common language which is used by websites to communicate with databases. Databases can be used to store persistent data, such as usernames and passwords, sensitive account data, or other important information used by the website. Databases are usually made of many "tables" 
-which are organised in rows and columns. Each row is a separate entry in the table, and each column is a specific parameter which can be used by the entry.
-
-As an example, a table in a database may look like this:
+### <a name="knowledge"></a>Background knowledge about SQL
+SQL (Structured Query Language) is a common language which is used by websites to communicate with databases. Databases can be used to store persistent data, such as usernames and passwords, sensitive account data, or other important information used by the website. Typically, SQL works on relational databases, which are usually made of many "tables" organised in rows and columns. Each row is a separate entry in the table, and each column is a specific parameter which can be used by the entry. A sample table is shown below:
 
 |   UserId   |    Username   |Password|
 |:----------:|:-------------:|:------:|
 |      1     |     Admin     | 123456 |
 |      2     |     Alice     | pw1234 |
 |      3     |      ...      |   ...  |
 
-
 SQL is used to interact with the database by sending "queries" which the database responds to. Some common SQL commands used include: 
 * `SELECT` - retrieves information from a table
 * `UPDATE` - changes information from a table
 * `INSERT INTO` - adds a new entry into a table
 * `DELETE FROM` - deletes information from a table
+* `DROP` - deletes the whole table
+
+In addition, a query could use parameters to filter, reorder, and group the returned results. For example, the query: 
+```SQL
+SELECT * FROM users WHERE Username = 'Admin'
+```
+will only returns the records in table "users" whose user name is "Admin". Here the parameter `Username = Admin` in the `WHERE clause` works as a filter.
+
+More information about SQL can be found [here](https://www.w3schools.com/sql/).
+
+### SQL Injection
+SQL injection is one of the most common web hacking techniques. It replaces malicious code in SQL statements via web page input. Websites combine SQL syntax with user input to query the underlying database of the website. However, some websites may not check the syntax of user input rigorously, and therefore a malicious user can inject SQL query via the user input.
+
+## Why is it Important to Prevent SQL Injection？
```

Comment:
This section comes too early in the article. Can be moved just above `Preventing SQL Injection` section.